### PR TITLE
[NPU] Add NPU to support hybrid engine

### PR DESCRIPTION
**Pull request by CurryRice233**
Add NPU to support the hybrid engine, we tested it with the example on [DeepSpeedExample](https://github.com/microsoft/DeepSpeedExamples/tree/master/applications/DeepSpeed-Chat/training/step3_rlhf_finetuning) and the results, average reward scores, were aligned with the GPU.

For works about NPU, see https://github.com/microsoft/DeepSpeed/issues/4567

### CODE REVIEW

**Comment by CurryRice233** [comment]
@mrwyattii hi, Could you please help to retrigger the workflow and review this PR ？

**Comment by CurryRice233** [comment]
@tjruwase @mrwyattii hi, would you please help to review this PR and add to merge queue? CI has passed.

**Comment by tjruwase** [comment]
@CurryRice233, thanks for this PR. Apologies for the delay. Most of the team are just returning from the holidays. We will review asap.

**Comment by CurryRice233** [comment]
@tjruwase The CI failed. It seems that there is a problem with the CI server. Can you check it?

**Comment by CurryRice233** [comment]
@tjruwase @loadams CI passed again, can you help to review this PR and add to merge queue?

**Comment by tjruwase** [comment]
> @tjruwase @loadams CI passed again, can you help to review this PR and add to merge queue?

@CurryRice233, we will review this asap!

File: op_builder/npu/inference.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+# DeepSpeed Team
+
+from enum import IntEnum
+import torch
+from .builder import NPUOpBuilder
+
+try:
+    import torch_npu
+except ImportError as e:
+    pass
```
Comment (mrwyattii) [inline op_builder/npu/inference.py:13]: @CurryRice233 importing torch here causes problems for DeepSpeed install. Can you please move the `torch` and `torch_npu` imports inside of a class method in this file? (See https://github.com/microsoft/DeepSpeed/blob/9500ab7d476aaf60ec2c76653c67bf5feb758f9e/op_builder/transformer_inference.py#L20 for an example)

The reason for this is that `op_builder` is imported by `setup.py` (before any requirements are installed) and if `torch` is not available the install will fail. Please let me know if you have any questions

**Review by mrwyattii — CHANGES_REQUESTED** [review]

File: op_builder/npu/inference.py (hunk 1 above)
Comment (CurryRice233) [reply op_builder/npu/inference.py:13]: Moving the `import torch' to the interior of the `try' solves this problem. If I move inside of class method, I need to import torch into each method, which I don't think is a good idea.

**Review by mrwyattii — APPROVED** [review]

**Comment by CurryRice233** [comment]
@mrwyattii  Hi, can you add this PR to merge queue, thank you !
